srecord/crc16.cc

```diff
@@ -240,10 +240,9 @@ srecord::crc16::updcrc(unsigned char c, unsigned short state)
     {
         return (((state >> 8) & 0xFF) | (c << 8)) ^ table[state & 0xFF];
     }
-    else
-    {
-        return ((state << 8) | c) ^ table[state >> 8];
```

Comment:
I'm not a fan of including this check or changing the code. As is, is more descriptive of intent. This is just pedantry from the linter. Happy to be convinced otherwise.

Reply:
I would also keep the if/else, but manually introduce a local variable to be returned at the end.
Coding style guides I know don't like early returns in the control flow, apart from error handling.
That way the linter is also happy.

Reply:
Perfect solution. Let's go with that